sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/AfterPane.java

```diff
@@ -100,6 +100,11 @@ public void clear(TriggerContext c) throws Exception {
   }
 
   @Override
+  public boolean hasState(TriggerContext c) throws Exception {
+    return !c.state().access(ELEMENTS_IN_PANE_TAG).isEmpty().read();
```

Comment:
Yes, I'm still thinking some 'trigger fired => no state other than finished bits' tests would be great. Will circle back to that, but note all the hasState stuff went away so I'll do it by looking at the underlying state table.